Not building on Flutter 3.10

There are 2 issues in the `deviceinfo_io.dart` from sdkgen preventing to build the app using Flutter 3.10.0

```
Launching lib/main.dart on macOS in debug mode...
main.dart:1
2023-05-10 20:45:48.187 xcodebuild[86921:36964624] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006000-000C792602D2801E }
{ platform:macOS, arch:x86_64, id:00006000-000C792602D2801E }
: Error: 'FlutterWindow' isn't a type.
deviceinfo_io.dart:15
      PlatformDispatcher.instance.views.whereType<FlutterWindow>().first;

                                                  ^^^^^^^^^^^^^
: Error: A value of type 'int?' can't be assigned to a variable of type 'String?'.
deviceinfo_io.dart:34
    platform['sdkVersion'] = androidInfo.version.sdkInt; // 29
                                                 ^
Target kernel_snapshot failed: Exception

Command PhaseScriptExecution failed with a nonzero exit code

** BUILD FAILED **

Exception: Build process failed
Exited
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -69,7 +69,7 @@ jobs:
           distribution: zulu
       - uses: subosito/flutter-action@v2
         with:
-          flutter-version: 3.3.x
+          flutter-version: 3.10.x
           channel: stable
       - run: cp LICENSE dart-runtime/
       - run: flutter pub get

--- a/.github/workflows/release_dev.yml
+++ b/.github/workflows/release_dev.yml
@@ -91,7 +91,7 @@ jobs:
           distribution: zulu
       - uses: subosito/flutter-action@v2
         with:
-          flutter-version: 3.3.x
+          flutter-version: 3.10.x
           channel: stable
       - run: cp LICENSE dart-runtime/
       - run: flutter pub get

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -66,7 +66,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        flutter-version: [3.3.x]
+        flutter-version: [3.10.x]
         flutter-channel: [stable]
     steps:
       - uses: actions/checkout@v3

--- a/dart-runtime/lib/deviceinfo_io.dart
+++ b/dart-runtime/lib/deviceinfo_io.dart
@@ -12,7 +12,7 @@ Future<Map<String, Object?>> getDeviceInfo(String deviceId) async {
   } catch (e) {}
 
   final window =
-      PlatformDispatcher.instance.views.whereType<FlutterWindow>().first;
+      PlatformDispatcher.instance.views.whereType<FlutterView>().first;
   final screenSize = window.physicalSize / window.devicePixelRatio;
 
   final platform = {

--- a/dart-runtime/pubspec.yaml
+++ b/dart-runtime/pubspec.yaml
@@ -4,16 +4,16 @@ homepage: https://sdkgen.github.io/
 description: Dart API client for sdkgen server. It must be used with a generated file, see docs.
 
 environment:
-  sdk: ">=2.18.0 <3.0.0"
-  flutter: ">=3.3.0"
+  sdk: '>=3.0.0 <4.0.0'
+  flutter: ">=3.10.0"
 
 dependencies:
   convert: ^3.1.1
   decimal: ^2.3.2
-  device_info_plus: ^8.0.0
+  device_info_plus: ^9.0.0
   equatable: ^2.0.0
   http: ^0.13.5
-  package_info_plus: ^3.0.2
+  package_info_plus: ^4.0.0
   shared_preferences: ^2.0.0
   flutter:
     sdk: flutter

